state-synchronizer/src/executor_proxy.rs

```diff
@@ -175,7 +175,7 @@ impl ExecutorProxyTrait for ExecutorProxy {
     }
 
     fn get_ledger_info(&self, version: u64) -> Result<LedgerInfoWithSignatures> {
-        let waypoint_li = self.storage.get_ledger_info(version)?;
+        let waypoint_li = self.storage.get_nearest_epoch_change_ledger_info(version)?;
         ensure!(
             waypoint_li.ledger_info().version() == version,
```

Comment:
Looking at the `ensure!()` here, the `version` should really be named `waypoint_version` or `epoch_change_version` or `epoch_boundary_version`, and it's expected to be a epoch_change ledger version at that version. I don't think we should even introduce `get_nearest_epoch_change_ledger_info()`, rather, how about let's just do a straightforward `.get_ledger_info_get_version() -> Result<Option<LI>>`, and ensure it returns a "Some" here.

Reply:
here it is. But it is also called in json-rpc method, which is another case.

Reply:
The other call site is a misuse, should not consider that. Can we get around it?